docs/CACHING.md

```diff
@@ -0,0 +1,353 @@
+
+First, make sure to install EBS CSI Driver in your cluster by following their [installation guide](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/install.md).
+
+You can then create a StorageClass using EBS CSI Driver for Mountpoint CSI Driver to request a volume to use as local cache:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: s3-cache-ebs-sc
+provisioner: ebs.csi.aws.com
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+parameters: # all optional, see https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/parameters.md for more details
+  type: io2
+  iopsPerGB: "256000"
+  blockExpress: "true"
+```
+
+You can then reference this StorageClass from your S3 PV:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv
+spec:
+  # ...
+  csi:
+    driver: s3.csi.aws.com
+    # ...
+    volumeAttributes:
+      bucketName: amzn-s3-demo-bucket
+      cache: ephemeral
+      cacheEphemeralStorageClassName: s3-cache-ebs-sc
+      cacheEphemeralStorageResourceRequest: 10Gi
+```
+
+With this configuration, once your workload is scheduled into a node, Mountpoint CSI Driver will schedule a Mountpoint Pod to the same node with the `ephemeral` volume. EBS CSI Driver will then dynamically provision an EBS volume and attach it to the node for Mountpoint to use as cache.
+
+The EBS volume and the Mountpoint Pod (therefore it's ephemeral PVC) will automatically cleaned up once the workload is terminated. We highly recommend you to use `reclaimPolicy: Delete` in your StorageClass to ensure the cache PV is automatically cleaned up as part of this process.
+
+##### Using Local Volume Static Provisioner to use local NVMe
+
+Some Amazon EC2 instances offer non-volatile memory express (NVMe) solid state drives (SSD) instance store volumes. You can utilise [Local Volume Static Provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner) to use instance store as cache. See [Instance store volume limits for EC2 instances](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-store-volumes.html) for more details about instance store support on EC2 instances, and [EKS Persistent Volumes for Instance Store](https://aws.amazon.com/blogs/containers/eks-persistent-volumes-for-instance-store/) on using instance storage in EKS.
+
+The Local Volume Static Provisioner allows you to configure some options, you can find more details in their [Getting started guide](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/docs/getting-started.md).
+
+As an example, you can configure your [eksctl](https://eksctl.io/) configuration to mount available NVMe instance storage disks at `/dev/disk/kubernetes`:
+
+```yaml
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+metadata:
+  name: cluster-with-storage
+  region: eu-central-1
+managedNodeGroups:
+  - name: storage-nvme
+    desiredCapacity: 1
```

Comment:
I would prefer to have working example for more than 1 EC2 Instance (e.g. 2) as it's far more popular use-case.